### ISSUE TITLE
Apply agent reporter standards to the Java agent

### ISSUE THREAD

**Issue by felixbarny** [elastic/apm-agent-java#64]
This document is an attempt to standardize on how backend APM agents buffer data before it’s sent to the APM Server and how the APM agents throttle requests to the APM Server and handle failures related to those requests:
https://docs.google.com/document/d/1a5O7lV1C77iVZSBDwTPbHC8nkiBu0mkaJy74WrVnX_Q/edit#heading=h.xw2nzvd1o6fr

The document is not finalized yet, so this is currently blocked by https://github.com/elastic/apm-server/issues/818

**Comment by alvarolobato** [comment]
This was already done when implementing Intake V2.
